### PR TITLE
Implement XpringClient

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Create a new class, `XpringClient`, which composes sub-services provided by Xpring.

### Context of Change

`XpringClient` will compose common interactions across Xpring.

Analog in JS: https://github.com/xpring-eng/Xpring-JS/pull/282
Analog in Java: https://github.com/xpring-eng/xpring4j/pull/179

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Easy `XpringClient` usage.

## Test Plan

New tests provided for CI.

### CODE REVIEW

File: XpringKit/Xpring/XpringClient.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  ///    - completion: A completion handler with the result of the operation.
+  /// - Throws: An error if there was a problem communicating with the XRP Ledger or the inputs were invalid.
+  // TODO(keefertaylor): Make this API synchronous to mirror functionality provided by ILP / XRP.
+  public func send(
+    _ amount: UInt64,
+    to destinationPayID: PaymentPointer,
+    from sourceWallet: Wallet,
+    completion: @escaping (Result<TransactionHash, Error>) -> Void
+  ) {
+    self.payIDClient.xrpAddress(for: destinationPayID) { [weak self] result in
```
Comment (amiecorso) [inline XpringKit/Xpring/XpringClient.swift:37]: just a question: is this `[weak self] .. ` line in the closure basically rendering `self` to refer to the `payIDClient` ?

File: Tests/Xpring/XpringClientTest.swift [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      switch result {
+      case .success(let transactionHash):
+        XCTAssertEqual(transactionHash, expectedTransactionHash)
+        completionCalledExpectation.fulfill()
+      case .failure:
+        XCTFail("Error making transaction")
+      }
+    }
+
+    self.wait(for: [completionCalledExpectation], timeout: 10)
```
Comment (hbergren) [inline Tests/Xpring/XpringClientTest.swift:59]: Nit, `// WHEN` without `// THEN`.

File: XpringKit.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -10,6 +10,7 @@
 		000EF674D02FE63D72B1E1C3 /* AccountBalance.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1585BA85CD7FF784C62F1885 /* AccountBalance.swift */; };
 		004CE447B535A5DDAAC7881D /* JavaScriptWalletFactory.swift in Sources */ = {isa = PBXBuildFile; fileRef = 6411247733F228E6454489A5 /* JavaScriptWalletFactory.swift */; };
 		02CE1498689341A9825BFA82 /* Rpc_V1_GetTxResponse+Test.swift in Sources */ = {isa = PBXBuildFile; fileRef = BB202B1B98F82FC7EEDA70F8 /* Rpc_V1_GetTxResponse+Test.swift */; };
+		02FFA28A80B08505F42BA007 /* XpringClient.swift in Sources */ = {isa = PBXBuildFile; fileRef = 161B7F046E7CC2194DC644CE /* XpringClient.swift */; };
```
Comment (hbergren) [inline XpringKit.xcodeproj/project.pbxproj:13]: I forget, what is the reason for checking in generated code?

File: XpringKit/PayID/PayIDError.swift [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /// Errors that originate from PayID.
-public enum PayIDError: Error {
+public enum PayIDError: Equatable, Error {
```
Comment (hbergren) [inline XpringKit/PayID/PayIDError.swift:2]: `Equatable`, Neat!

Do we not need to implement a custom compare function for this to work?

File: XpringKit/Xpring/XpringClient.swift (hunk 1 above)
Comment (hbergren) [reply XpringKit/Xpring/XpringClient.swift:37]: Swift learning opportunity: What is `[weak self]`?

Comment (keefertaylor) [reply XpringKit/Xpring/XpringClient.swift:37]: Talked offine. This is a capture list that is preventing a retain cycle.

File: XpringKit.xcodeproj/project.pbxproj (hunk 3 above)
Comment (keefertaylor) [reply XpringKit.xcodeproj/project.pbxproj:13]: Cathage builds via checking out a github repo.

File: XpringKit/PayID/PayIDError.swift (hunk 4 above)
Comment (keefertaylor) [reply XpringKit/PayID/PayIDError.swift:2]: Yeah, Swift is able to synthesize this on enums.

**Comment by keefertaylor** [comment]
> Swift learning opportunity: What is [weak self]?

Somewhat of a loaded question :) Basically, this closure is capturing references and the garbage collector in swift works via reference counting. So `self` (`this` in JS/Java) has a strong reference to the closure, and the closure has a strong reference to `self`. This creates a "retain cycle" where the garbage collector can't deallocate either and we end up with memory leaks. 

I'm going to link you out to an article because you kinda need fancy diagrams to fully grok what's happening here but I am happy to follow up with anything that is unclear: 
https://medium.com/mackmobile/avoiding-retain-cycles-in-swift-7b08d50fe3ef

**Pull request merged**